helm-v3/001-charts.md

```diff
@@ -215,3 +219,43 @@ This can be schematized into something like this:
 In many cases, the base types in a YAML file can be used to derive a base
 schema. Thus it may be desirable to automatically derive schemata if they are
 not supplied.
+
+## Version Reference System
+
+In order to make things look and feel more like Docker, which many people have become comfortable with,
+the `--version` flag will be removed on all subcommands (with the exception of `helm package`).
+
+Referencing a specific chart version can be achieved using a new version reference system.
+Very simply, a chart version can be specified using `NAME[:VERSION]`.
+
+The following commands in Helm 2:
+
+```
+helm fetch stable/hackmd --version 0.1.0
+helm search stable/hackmd --version 0.1.0
+helm inspect stable/hackmd --version 0.1.0
+helm install stable/hackmd --version 0.1.0
+helm upgrade hackmd stable/hackmd --version 0.1.0
+```
+
+Will instead look like the following in Helm 3:
+
+```
+helm pull stable/hackmd:0.1.0
+helm search stable/hackmd:0.1.0
+helm inspect stable/hackmd:0.1.0
+helm install stable/hackmd:0.1.0
+helm upgrade hackmd stable/hackmd:0.1.0
+```
+
+If no version is specified, the latest stable semantically-versioned chart in the repository will be used by default
+(same as it works in Helm 2).
+
+Immutability, or "releases" as defined by
+[App Registry](https://github.com/app-registry/appr/blob/master/Documentation/quick-start.md),
+can be added at a later date without introducing any breaking changes by updating the version reference system to
+accept additional characters (e.g. `@1.0.0`).
```

Comment:
what other characters besides '-' are needed? Ex  4.0.0-alpha1 
I can't think of a reason why you would want to type for example, `helm install stable/chart-~1.0.1` when you know that chart-1.0.1 is stable for your needs. Unless I just totally read this wrong.

Reply:
@stefsuse take a look at what app registry does in regards to releases vs channels: https://github.com/app-registry/appr/blob/master/Documentation/quick-start.md

By other characters, I mean separators between name and version that are not `:` (e.g. `@` to specify an immutable release)

What comes after the `:` character is the chart version, and that format will comply with whatever version validation is already in place